### PR TITLE
Export to ONNX doc refocused on using optimum, added tflite

### PR DESCRIPTION
**Pull request by MKhalusova**
Currently, despite the notes saying that the recommended way is to use Optimum, the main focus of the doc is on using `transformers.onnx`, which is no longer maintained (according to optimum team). 
This PR restructures the doc to put the optimum examples forward as the primary way for exporting models to ONNX. 
The example of using `transformers.onnx` is kept for potential compatibility reasons, however, I have removed the examples of adding new architectures to transformers.onnx, as I believe this should be done in Optimum instead (links provided).
As suggested by the team, I have also added an example for TFLite, and for that reason, renamed the page to "Export to ONNX, TFLite" 

UPD: since we now link to Optimum docs for the list of supported architectures, I have also removed the ONNX list check & update from the check_table script.

### CODE REVIEW

File: docs/source/en/serialization.mdx [hunk 1]
```diff
@@ -127,416 +121,183 @@ Ready-made configurations include the following architectures:
...
-
-* Encoder-based models inherit from [`~onnx.config.OnnxConfig`]
-* Decoder-based models inherit from [`~onnx.config.OnnxConfigWithPast`]
-* Encoder-decoder models inherit from [`~onnx.config.OnnxSeq2SeqConfigWithPast`]
-
-<Tip>
-
-A good way to implement a custom ONNX configuration is to look at the existing
-implementation in the `configuration_<model_name>.py` file of a similar architecture.
+`tranformers.onnx` is no longer maintained, please export models with 🤗 Optimum as described above. This section is will be removed in the future versions.
```
Comment (regisss) [inline docs/source/en/serialization.mdx:225]:
```suggestion
`tranformers.onnx` is no longer maintained, please export models with 🤗 Optimum as described above. This section will be removed in the future versions.
```

File: docs/source/en/serialization.mdx [hunk 2]
```diff
@@ -127,416 +121,183 @@ Ready-made configurations include the following architectures:
...
-
-## Exporting a model for an unsupported architecture
-
-<Tip>
+### Exporting a model for an unsupported architecture
 
 If you wish to contribute by adding support for a model that cannot be currently exported, you should first check if it is
-supported in [`optimum.exporters.onnx`](https://huggingface.co/docs/optimum/main/en/exporters/onnx/package_reference/configuration#supported-architectures),
-and if it is not, [contribute to 🤗 Optimum](https://huggingface.co/docs/optimum/main/en/exporters/onnx/usage_guides/contribute)
+supported in [`optimum.exporters.onnx`](https://huggingface.co/docs/optimum/exporters/onnx/package_reference/configuration#supported-architectures),
```
Comment (regisss) [inline docs/source/en/serialization.mdx:217]: This section is not there anymore it seems, so let's point to this other part of the doc where all the supported architectures are listed:
```suggestion
supported in [`optimum.exporters.onnx`](https://huggingface.co/docs/optimum/exporters/onnx/overview),
```

File: docs/source/en/serialization.mdx [hunk 3]
```diff
@@ -127,416 +121,183 @@ Ready-made configurations include the following architectures:
...
-  [`~onnx.features.FeatureManager`]
-* Add your model architecture to the tests in `test_onnx_v2.py`
+```bash
+optimum-cli export tflite --model bert-base-uncased --sequence_length 128 bert_tflite/
+```
 
-Check out how the configuration for [IBERT was
-contributed](https://github.com/huggingface/transformers/pull/14868/files) to get an
-idea of what's involved.
+Call `optimum-cli export tflite --help` to review the full list of supported arguments.
```
Comment (regisss) [inline docs/source/en/serialization.mdx:303]: I would also point to the list of currently supported arcitectures [here](https://huggingface.co/docs/optimum/exporters/tflite/overview).

**Review by regisss — COMMENTED** [review]
Awesome work @MKhalusova :fire: :rocket: 

My main comment is about whether we should have a dedicated page for TFLite to give it more visibility. Then, since there is much less to say and its support in Optimum is still in its infancy, I'm also fine with revisiting this question later when we will have more content to write :slightly_smiling_face:

File: docs/source/en/_toctree.yml [hunk 4]
```diff
@@ -92,7 +92,7 @@
     - local: sagemaker
       title: Run training on Amazon SageMaker
     - local: serialization
-      title: Export to ONNX
+      title: Export to ONNX, TFLite
```
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/en/_toctree.yml:95]: Sure, we can split this into two pages and add more content when there's more functionality

File: docs/source/en/serialization.mdx [hunk 5]
```diff
@@ -27,27 +33,15 @@ By exposing a graph with standardized operators and data types, ONNX makes it ea
...
+- run with [optimized inference pipelines](https://huggingface.co/docs/optimum/main/en/onnxruntime/usage_guides/pipelines),
+which has the same API as the [`pipeline`] function in 🤗 Transformers. 
 
-To explore all these features,  check out the [🤗 Optimum library](https://github.com/huggingface/optimum).
-
-</Tip>
+🤗 Optimum provides support for the ONNX export by leveraging configuration objects. These configuration objects come 
+ready-made for a number of model architectures, and are designed to be easily extendable to other architectures.
 
 Ready-made configurations include the following architectures:
```
[the remark this answers is not in the document]
Comment (MKhalusova) [reply docs/source/en/serialization.mdx]: I removed the list from the doc which made CI fail. The list was built automatically, so I have now also removed the part of script that checks the list and builds it.

**Review by regisss — APPROVED** [review]
Thanks @MKhalusova, it looks great!

**Review by amyeroberts — APPROVED** [review]
Changes all look good to me - thanks for adding and tidying up! 

Pinging @sgugger to double check there's no additional dependencies we need to consider in relation to the removal of the ONNX table?

**Review by sgugger — APPROVED** [review]
Thanks for updating this guide. All good for the removal of the ONNX table.

**Pull request merged**